package-res/resources/web/pentaho/visual/role/i18n/messages.properties

```diff
@@ -0,0 +1,51 @@
+## Property
+property.label=Visual role property
+property.description=Holds a visual role mapping.
+
+## QuantitativeMapping
+quantitative.label=Quantitative visual role mapping
+
+## OrdinalMapping
+ordinal.label=Ordinal visual role mapping
+
+## NominalMapping
+nominal.label=Nominal visual role mapping
+
+## Mapping
+mapping.label=Visual role mapping
+mapping.description=The mapping between a visual role and data properties.
+mapping.props.level.label=Fixed level of measurement
+mapping.props.attrs.label=Attributes
+
+## MappingAttribute
+mappingAttribute.label=Visual role mapping attribute
+mappingAttribute.description=A data property in a visual role mapping.
+mappingAttribute.props.name.label=Property name
+mappingAttribute.props.aggr.label=Aggregation
+mappingAttribute.props.isReverse.label=Reverse natural order
+
+## Aggregation
+aggregation.domain.sum.f=Sum
+aggregation.domain.avg.f=Average
+aggregation.domain.min.f=Minimum
+aggregation.domain.max.f=Maximum
+
+## Measurement Level
+level.domain.nominal.f=Nominal
+level.domain.ordinal.f=Ordinal
+level.domain.quantitative.f=Quantitative
+
+## Errors
+errors.mapping.dataTypeNotSubtypeOfBaseType=Visual role mapping subtypes must have a 'dataType' attribute that is a subtype of the base mapping's 'dataType'.
+errors.mapping.dataTypeIncompatibleWithRoleLevel=Qualitative data type '{0}' is not compatible with quantitative measurement level '{1}'.
+errors.mapping.levelsLockedWhenTypeHasDescendants=Cannot change the 'levels' attribute of a Mapping that has descendants.
+errors.mapping.dataTypeLockedWhenTypeHasDescendants=Cannot change the 'dataType' attribute of a Mapping that has descendants.
+
+## Validation Errors
+errors.mapping.levelIsNotOneOfRoleLevels=The mapping of visual role '{role}' has a fixed level of measurement '{level}' which is not one of the role's levels {roleLevels}.
+errors.mapping.noOwnerVisualModel=The visual role mapping must have an owner visual model.
+errors.mapping.attributeIsNotDefinedInVisualModelData=The attribute '{name}' in visual role '{role}' is not defined in the data source.
+errors.mapping.attributeDataTypeNotSubtypeOfRoleType=The data type of the attribute '{name}', '{dataType}', is not a subtype of '{roleDataType}', the data type required by the visual role '{role}'.
+errors.mapping.attributesLevelNotCompatibleWithRoleLevels=The attribute in visual role '{role}' have an effective level of measurement '{dataLevel}' which is not compatible with the visual roles' levels {roleLevels}.
```

Comment:
Suggest:

The attribute in visual role '{role}' has an effective level of measurement '{dataLevel}' that is not compatible with the visual roles' levels {roleLevels}.